fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/lock/InMemoryResourceLockManager.java

```diff
@@ -29,67 +38,114 @@ public class InMemoryResourceLockManager implements ResourceLockManager {
 
     private static final Logger LOG = LoggerFactory.getLogger(InMemoryResourceLockManager.class);
 
-    private final Map<String, Set<String>> transactionLocks;
-    private final Set<String> lockedResources;
+    private final Map<String, Set<ResourceLock>> transactionLocks;
+    private final Map<String, Set<ResourceLock>> resourceLocks;
```

Comment:
I think it would be fine to use FedoraID as the key for resourceLocks, it implements `equals` and `hashCode`, and the value used for them is immutable. Just putting that suggestion out there if you'd like to cut down on the `getResourceId().getResourceId()` calls.